src/main/java/build/buildfarm/instance/server/AbstractServerInstance.java

```diff
@@ -326,7 +329,15 @@ public DigestUtil getDigestUtil() {
             directExecutor()));
   }
 
-  private static boolean shouldEnsureOutputsPresent(RequestMetadata requestMetadata) {
+  private static boolean shouldEnsureOutputsPresent(
+      boolean ensureOutputsPresent, RequestMetadata requestMetadata) {
+
+    // We will ensure outputs are present if the system is globally configured to check.
+    // Otherwise we will determine if dynamically from optional URI parameters.
+    if (ensureOutputsPresent) {
```

Comment:
I don't think we should ignore the ENSURE_OUTPUTS_PRESENT but rather check for the validity of both. So if ensureOutputsPresent AND ENSURE_OUTPUTS_PRESENT exists in the url then we should perform this check. This will ensure that we can set the global config but only perform the check if the build specifically requests to use bwob.

Reply:
Unless the intent here is to do global INSTEAD of passing ENSURE_OUTPUTS_PRESENT?

Reply:
Yeah, use case is if you can't ass in `requestMetadata` or don't want to set it for every request. I think it's also nice to save ourselves time in parsing URI.

Reply:
improved comment